optuna/samplers/_tpe/parzen_estimator.py

```diff
@@ -185,29 +185,33 @@ def _calculate_categorical_distributions(
         search_space: CategoricalDistribution,
         parameters: _ParzenEstimatorParameters,
     ) -> _BatchedDistributions:
-        consider_prior = parameters.consider_prior or len(observations) == 0
+        choices = search_space.choices
+        n_choices = len(choices)
+        n_samples = observations.size + (parameters.consider_prior or observations.size == 0)
```

Comment:
Nit: the name `n_samples` might not be appropriate, because this includes prior. Maybe `n_mixture` may be better?